### PR TITLE
Create the Windows.x64 global tool with shim for signing

### LINKED ISSUE #20557: pwsh.exe shim from .NET tool install is not signed
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

`dotnet tool install --global PowerShell`

### Expected behavior

```console
`%USERPROFILE%\.dotnet\tools\pwsh.exe` should be a signed file

This is particularly needed for the .NET SDK container images which [install PowerShell as a global tool](https://github.com/dotnet/dotnet-docker/blob/6ba62d6825b9f45b3d665ab6cc930808a40d63c6/src/sdk/8.0/nanoserver-ltsc2022/amd64/Dockerfile#L39-L47).
```

### Actual behavior

```console
`%USERPROFILE%\.dotnet\tools\pwsh.exe` is unsigned
```

### Error details

_No response_

### Environment data

```powershell
PowerShell 7.3.8
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Fixes https://github.com/PowerShell/PowerShell/issues/20557

The global tool for Windows.x64 is created using dotnet pack so that shim can be generated during the build for signing.
The tool uses a new csproj with added properties for generating the shim, we added the build to fxdependent stage.

More details about shim are documented here: https://github.com/dotnet/sdk/blob/main/documentation/general/signing-global-tool-packages.md

The way the build works is a bit tricky as when it is built, we cannot add the modules that come from gallery.
We have to use `dotnet pack` to build the nupkg and then expand it as a zip.
After expanding we restore the signed files for the modules from the gallery.
We also delete `content` and `contentFiles` folder which are not necessary.
After that, we repack using Compress-Archive and rename it back to a nupkg.

## PR Context

dotnet-sdk images show that the exe is unsigned and fails compliance requirements.

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: .pipelines/templates/obp-file-signing.yml [hunk 1]
```diff
@@ -138,10 +139,17 @@ steps:
 - pwsh: |
     Import-Module '$(PowerShellRoot)/build.psm1' -Force
     Import-Module '$(PowerShellRoot)/tools/packaging' -Force
-    $pathForUpload = New-Item -ItemType Directory -Path '$(ob_outputDirectory)/Signed-$(Runtime)' -Force
-    Write-Verbose -Verbose -Message "pathForUpload: $pathForUpload"
-    Copy-Item -Path '${{ parameters.binPath }}\*' -Destination $pathForUpload -Recurse -Force -Verbose
-    Write-Verbose -Verbose -Message "Files copied to $pathForUpload"
+    $isGlobalTool = '${{ parameters.globalTool }}' -eq 'true'
+
```
Comment (adityapatwardhan) [inline .pipelines/templates/obp-file-signing.yml:143]: Extra space

Comment (adityapatwardhan) [inline .pipelines/templates/obp-file-signing.yml:143]:
```suggestion
```

File: .pipelines/templates/obp-file-signing.yml [hunk 2]
```diff
@@ -138,10 +139,17 @@ steps:
 - pwsh: |
     Import-Module '$(PowerShellRoot)/build.psm1' -Force
     Import-Module '$(PowerShellRoot)/tools/packaging' -Force
-    $pathForUpload = New-Item -ItemType Directory -Path '$(ob_outputDirectory)/Signed-$(Runtime)' -Force
-    Write-Verbose -Verbose -Message "pathForUpload: $pathForUpload"
-    Copy-Item -Path '${{ parameters.binPath }}\*' -Destination $pathForUpload -Recurse -Force -Verbose
-    Write-Verbose -Verbose -Message "Files copied to $pathForUpload"
+    $isGlobalTool = '${{ parameters.globalTool }}' -eq 'true'
```
Comment (TravisEz13) [inline .pipelines/templates/obp-file-signing.yml:142]: Not blocking.  We should move any PS Script of significant size into a script or module.  This does two things, makes sure they are scanned by static analysis tools and reduces total yaml size.

File: tools/packaging/packaging.psm1 [hunk 3]
```diff
@@ -4341,20 +4341,22 @@ function New-GlobalToolNupkgSource
...
-            Copy-Item -Path $iconPath -Destination "$RootFolder/$iconFileName" -Verbose
-
-            $ridFolder = New-Item -Path (Join-Path $RootFolder "tools/$script:netCoreRuntime/any") -ItemType Directory
-
-            Write-Log "New-GlobalToolNupkgSource: Copying runtime assemblies from $WindowsDesktopBinPath for $PackageType"
-            Copy-Item "$WindowsDesktopBinPath/*" -Destination $ridFolder -Recurse
-            Remove-Item -Path $ridFolder/runtimes/win-arm -Recurse -Force
-            $toolSettings = $packagingStrings.GlobalToolSettingsFile -f "pwsh.dll"
-        }
+        # Due to needing a signed shim for the global tool, we build the global tool in build instead of packaging.
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1:4344]: remove dead code?

**Review by TravisEz13 — APPROVED** [review]
Approved with comments

File: tools/packaging/packaging.psm1 (hunk 3 above)
Comment (adityapatwardhan) [reply tools/packaging/packaging.psm1:4344]: I wanted to keep the code and document why we don't build Windows.x64 here.

**Pull request merged**